openpype/hosts/maya/plugins/publish/extract_playblast.py

```diff
@@ -65,12 +68,44 @@ def process(self, instance):
         # get cameras
         camera = instance.data["review_camera"]
 
-        preset = lib.load_capture_preset(data=self.capture_preset)
-        # Grab capture presets from the project settings
-        capture_presets = self.capture_preset
+        host_name = instance.context.data["hostName"]
+        family = instance.data["family"]
+        task_data = instance.data["anatomyData"].get("task", {})
+        task_name = task_data.get("name")
+        task_type = task_data.get("type")
+        subset = instance.data["subset"]
+
+        filtering_criteria = {
+            "hosts": host_name,
+            "families": family,
+            "task_names": task_name,
+            "task_types": task_type,
+            "subset": subset
+        }
+
+        if self.profiles:
+            profile = filter_profiles(
+                self.profiles, filtering_criteria, logger=self.log
+            )
+            capture_preset = profile.get("capture_preset")
+        else:
+            self.log.warning("No profiles present for Extract Playblast")
+
+        # Backward compatibility for deprecated Extract Playblast settings
+        # without profiles.
+        if capture_preset is None:
+            capture_preset = self.capture_preset
+
+        preset = lib.load_capture_preset(data=capture_preset)
```

Comment:
Should this maybe be moved to `lib`? It already appears to be duplicated and has quite some lines of code. By moving to lib it can be re-used across the creator and extractors like:

```python
def get_capture_profile(host_name, family, task_name, task_type, subset, project_settings=None):
    
    if not project_settings:
        project_name = get_current_project_name()
        project_settings = get_project_settings(project_name)

    playblast_settings = project_settings["maya"]["publish"]["ExtractPlayblast"]
    profiles = playblast_settings.get("profiles")
    preset = None
    if profiles:
        asset_doc = get_asset_by_name(project_name, data["asset"])
        task_name = get_current_task_name()
        task_type = asset_doc["data"]["tasks"][task_name]["type"]

        filtering_criteria = {
            "hosts": "maya",
            "families": "review",
            "task_names": task_name,
            "task_types": task_type,
            "subset": data["subset"]
        }
        profile = filter_profiles(
            profiles, filtering_criteria, logger=self.log
        )
        
        if profile is not None:
            return profile
        else:
            log.warning(
                "No profile for Extract Playblast capture"
                "with criteria: {}".format(
                    json.dump(filtering_criteria, indent=4)
                )
            )
            # TODO Do we want to raise an error here?
    else:
        # Backward compatibility for deprecated Extract Playblast settings
        # without profiles.
        log.debug("Falling back to deprecated Extract Playblast capture preset "
                  "because no new style playblast profiles are defined")
        return playblast_settings["capture_preset"]
```

That way you don't need to duplicate the backwards compatibility logic.

You could use it in the extractors like:

```python
task_data = instance.data["anatomyData"].get("task", {})
context = instance.context
preset = get_capture_profile_preset(
        host_name=context.data["hostName"],
        families=instance.data["family"],
        task_name=task_data.get("name"),
        task_type=task_data.get("type"),
        subset=instance.data["subset"],
        project_settings=context.data["project_settings"]
)
```

Note that it still can return None if profiles are defined but there's no match for the current profile criteria.

Reply:
Yeah good idea.